vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -35,6 +36,17 @@ class Action(VizroBaseModel):
         regex="^[a-zA-Z0-9_]+[.][a-zA-Z_]+$",
     )
 
+    # TODO: Abstract away once pattern becomes clear regarding function/import checks
+    @validator("function")
+    def check_export_imports(cls, function):
+        if (
+            function._CapturedCallable__function.__name__ == "export_data"
+            and function._arguments.get("file_format") == "xlsx"
```

Comment:
```suggestion
            function._function.__name__ == "export_data"
            and function._arguments.get("file_format") == "xlsx"
```

The `CapturedCallable._function` property is there to avoid needing to access the name-mangled `__function`. Accessing something that is name-mangled like that is very unlikely to be the right thing to do, even in a private function.

I don't much like the use of `function._arguments.get` (as per the comments on the `_arguments` property I hope to remove it some time if feasible) but I see why you've done it since I know `function.get` won't itself work. 

We should probably [`apply_defaults`](https://docs.python.org/3/library/inspect.html#inspect.BoundArguments.apply_defaults) in `CapturedCallable.__getitem__` or `CapturedCallable.__init__` to get things like this working more nicely, but that's way outside the scope of this PR and would need some extra thought and testing, so let's not do here.

Reply:
Ah good point! Yes it didn't feel right at all! Just had forgotten about how to do it the proper way. Thanks!

Suggestion makes sense, shall I leave a note here? I guess it will be found automatically once we remove it, so maybe not necessary

Reply:
I see we also have a `TODO` on the one other call to `_arguments`, so let's put one in here as well just so we've got it everywhere a developer might be looking.

Reply:
Added it to the `TODO` in `_arguments`, there was no `TODO` in `CapturedCallable.__getitem__`, so I left that one out.